pkg/contexts/ocm/repositories/composition/repository.go

```diff
@@ -178,7 +191,11 @@ func (v *VersionAccess) Close() error {
 }
 
 func (v *VersionAccess) IsReadOnly() bool {
-	return false
+	return v.readonly
+}
+
+func (v *VersionAccess) SetReadOnly() {
+	v.readonly = true
```

Comment:
As discussed... this is now a one-way action!!
Once things are read-only, there is no way to revert and make it changeable again. From pure API perspective it looks a bit odd.

@mandelsoft - can you please document this and elaborate the details, why this is the right way to do?